src/notte/common/notifier/base.py

```diff
@@ -0,0 +1,50 @@
+from abc import ABC, abstractmethod
+
+from notte.common.agent.base import BaseAgent
+from notte.common.agent.types import AgentResponse
+
+
+class BaseNotifier(ABC):
+    """Base class for notification implementations."""
+
+    @abstractmethod
+    async def send_message(self, text: str) -> None:
+        """Send a message using the specific notification service."""
+        pass
+
+    async def notify(self, task: str, result: AgentResponse) -> None:
```

Comment:
can you add a  `def format_message` function instead of having this inside `notitif` in that way you can override this function in the email client to avoid code duplication (cf next comment)